deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -559,17 +580,22 @@ async def get_interactive_waveform_data(
     upload_id = str(upload_id)
     job_id = str(job_id)
 
-    # grab jobs meta data to use in peak_detector if necessary
-    # TODO remove this step when peaks and valleys get saved to parquet on initial analysis
     try:
         async with request.state.pgpool.acquire() as con:
             logger.info(f"Getting metadata for job {job_id}")
             jobs = await _get_jobs(con, token, [job_id])
 
         selected_job = jobs[0]
         parsed_meta = json.loads(selected_job["job_meta"])
-        analysis_params = parsed_meta["analysis_params"]
         recording_owner_id = str(selected_job["user_id"])
+        pulse3d_version = parsed_meta.get("version", None)
+
+        # only compatible with Pulse3D 0.28.0 and above because of changes in pulse3d-worker
+        # interactive analysis was initially compatible with 0.25.2 and above, but pulse3d-worker is not backwards compatible
+        if pulse3d_version is None or VersionInfo.parse(pulse3d_version) < "0.28.0":
+            return NoJobDataFoundResponse(
```

Comment:
We probably want this to return some kind of 4xx error code, I don't think it does that right now. Also once we do that I'm not sure if we need to have an entire Model just to return an error message